utilities/ideascale-importer/ideascale_importer/reviews_importer.py

```diff
@@ -0,0 +1,96 @@
+from lxml import html
+import secrets
+
+from ideascale_importer import utils
+
+class Client:
+    """IdeaScale front-end API client."""
+
+    DEFAULT_API_URL = "https://cardano.ideascale.com"
+
+    def __init__(self, ideascale_url):
+        self.inner = utils.HttpClient(ideascale_url)
+
+    async def close(self):
+        await self.inner.close()
+
+    async def login(self,  email, password):
+        login = "/a/community/login"
+
+        data = {
+            'target-content-type': 'json',
+            'emailAddress': email,
+            'password': password,
+            'rememberMe': 'true',
+        }
+
+        await self.inner.post(f"{login}", data=data)
+
+    async def download_reviews(self, funnel_id, out_dir):
+        async def download_file(self, funnel_id, id, out_dir):
+            export_endpoint = "/a/admin/workflow/survey-tools/assessment/report/statistic/export/assessment-details/"
+            file_name = f"{funnel_id}_{secrets.token_hex(6)}"
+
+            content = await self.inner.get(f"{export_endpoint}{id}") 
+            tree = html.fromstring(content)
+
+            # we are looking for '<div class="card panel export-result-progress" data-features="refresh-processing-item" data-processing-item-id="15622">'
+            # to get "15622" value as for example
+            item  = int(tree.find(".//div[@data-processing-item-id]").get("data-processing-item-id"))
+
+            export_data_endpoint = "/a/reporting/export-data/"
+            while True:
+                content = await self.inner.get(f"{export_data_endpoint}{item}")
+                if "Finished Processing" in str(content):
+                    download_endpoint = "/a/download-export-file/"
+
+                    content = await self.inner.get(f"{download_endpoint}{item}")
+                    f = open(f"{out_dir}/{file_name}.xls", "wb")
```

Comment:
Instead of using temporary files, it would be better to actually read the excel data here as its read, and convert it.
You can use https://openpyxl.readthedocs.io/en/stable/api/openpyxl.reader.excel.html to read excel files.
And `io.BytesIO(content)` to convert the content downloaded into an in-memory file like object which the openpyxl reader can handle.
This will also tell you right here if there is an issue with the data being downloaded or not.